[WTF] WTF_ALWAYS_LOG

mdubet opened the pull request:
#### f6491fe8a5f06df095ea8abaa02ff8ad68c66c63
<pre>
[WTF] WTF_ALWAYS_LOG
<a href="https://bugs.webkit.org/show_bug.cgi?id=270115">https://bugs.webkit.org/show_bug.cgi?id=270115</a>

Reviewed by Simon Fraser.

No more &quot;stream &lt;&lt;&quot; typing everytime.

* Source/WTF/wtf/Assertions.h:

Canonical link: <a href="https://commits.webkit.org/276429@main">https://commits.webkit.org/276429@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/ee76487b950eab12814f5eef58d25785563799e4

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/41599 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/20613 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/43979 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/44168 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/37691 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/23723 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/17943 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/34397 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/42173 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/17534 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/35831 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/15061 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/15245 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/36840 "Passed tests") | [⏳ 🛠 wpe-skia ](https://ews-build.webkit.org/#/builders/WPE-Skia-Build-EWS "Waiting in queue, processing has not started yet") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/35052 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/37781 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/37161 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/45541 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/41225 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/16414 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/13479 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/40915 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/18033 "Built successfully") | | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/39330 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/9941 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/18089 "Built successfully") | | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/48236 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/17677 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/9839 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/ee76487b950eab12814f5eef58d25785563799e4. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/25131

rreno commented:
I'd like to add a vote to the idea to replace `ALWAYS_LOG_WITH_STREAM` with this by removing the old and renaming this macro you're adding to `WTF_ALWAYS_LOG_WITH_STREAM`.

mdubet commented:
Is the `WITH_STREAM` really helpful ?

stwrt commented on Source/WTF/wtf/Assertions.h:
```diff
@@ -681,6 +681,12 @@ constexpr bool assertionFailureDueToUnreachableCode = false;
         WTFLogAlways("%s", stream.release().utf8().data()); \
     } while (0)
 
+#define WTF_ALWAYS_LOG(commands) do { \
```
It would be good to replace the `ALWAYS_LOG_WITH_STREAM` macro, so we don't have two competing implementations.

There are only a handful uses of `ALWAYS_LOG_WITH_STREAM` when performing a code search.

mdubet replied:
I suggest to let it there a bit for a transition period, and we can remove in a future patch ?

stwrt commented:
> Is the `WITH_STREAM` really helpful ?

I am not convinced the WITH_STREAM is helpful.

mdubet committed:
[WTF] WTF_ALWAYS_LOG
https://bugs.webkit.org/show_bug.cgi?id=270115

Reviewed by Simon Fraser.

No more "stream <<" typing everytime.

* Source/WTF/wtf/Assertions.h:

Canonical link: https://commits.webkit.org/276429@main

webkit-commit-queue commented:
Committed 276429@main (f6491fe8a5f0): <https://commits.webkit.org/276429@main>

Reviewed commits have been landed. Closing PR #25131 and removing active labels.
